chain/xrp/client/client.go

```diff
@@ -0,0 +1,718 @@
+	txInput.LastLedgerSequence = lastLedgerSequence
+
+	return *txInput, nil
+}
+
+// FetchTransferInput returns tx input for a Template tx
+func (client *Client) FetchTransferInput(ctx context.Context, args xcbuilder.TransferArgs) (xc.TxInput, error) {
+	txInput, err := client.FetchBaseInput(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &txInput, nil
+}
+
+// Deprecated method - use FetchTransferInput
+func (client *Client) FetchLegacyTxInput(ctx context.Context, from xc.Address, to xc.Address) (xc.TxInput, error) {
+	// No way to pass the amount in the input using legacy interface, so we estimate using min amount.
+	args, _ := xcbuilder.NewTransferArgs(from, to, xc.NewAmountBlockchainFromUint64(1))
+	txInput, err := client.FetchTransferInput(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return txInput, nil
+}
+
+// SubmitTx submits a Template tx
+func (client *Client) SubmitTx(ctx context.Context, txInput xc.Tx) error {
+	serializedTxInput, err := txInput.Serialize()
+	if err != nil {
+		return err
+	}
+
+	submitRequest := &SubmitRequest{
+		Method: Submit,
+		Params: []SubmitParamEntry{
+			{
+				TxBlob: string(serializedTxInput),
+			},
+		},
+	}
+
+	var submitResponse SubmitResponse
+	err = client.Send(MethodPost, submitRequest, &submitResponse)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FetchTxInfo Returns transaction info - new endpoint
+func (client *Client) FetchTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	legacyTxInfo, err := client.FetchLegacyTxInfo(ctx, txHash)
+	if err != nil {
+		return xclient.TxInfo{}, err
+	}
+
+	// Remap to new tx
+	return xclient.TxInfoFromLegacy(client.Asset.GetChain().Chain, legacyTxInfo, xclient.Account), nil
+}
+
+// FetchLegacyTxInfo Returns transaction info - legacy/old endpoint
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xc.LegacyTxInfo, error) {
+
+	txRequest := &TransactionRequest{
+		Method: Transaction,
+		Params: []TransactionParamEntry{
+			{
+				Transaction: txHash,
+				Binary:      false,
+			},
+		},
+	}
+
+	var txResponse TransactionResponse
+	err := client.Send(MethodPost, txRequest, &txResponse)
+	if err != nil {
+		return xc.LegacyTxInfo{}, err
+	}
+
+	ledgerRequest := LedgerRequest{
+		Method: Ledger,
+		Params: []LedgerParamEntry{
+			{
+				LedgerIndex: "current",
+			},
+		},
+	}
+
+	var ledgerResponse LedgerResponse
+	err = client.Send(MethodPost, ledgerRequest, &ledgerResponse)
+	if err != nil {
+		return xc.LegacyTxInfo{}, err
+	}
+
+	confirmations := ledgerResponse.Result.LedgerCurrentIndex - txResponse.Result.Sequence
+
+	explorer := client.Asset.GetChain().ExplorerURL + "/tx/" + txResponse.Result.Hash + "?cluster=" + client.Asset.GetChain().Net
+
+	var sources []*xc.LegacyTxInfoEndpoint
+	sources = append(sources, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.Amount),
```

Comment:
contract address? for token amounts

Reply:
I thought address is the source/destination address.

Reply:
But there is also a field named ContractAddress which I think it is as the names implies. So Address might be the wallet address.